<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jason Lerch and Kamila Szulc-Lerch">
<meta name="dcterms.date" content="2025-02-08">
<meta name="description" content="You want to run a four core genotype analysis without having all four genotypes">

<title>Unbalanced Four Core Genotype Analyses – digressions on (mouse) brain imaging</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-K98RZYWMVC"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-K98RZYWMVC', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<meta name="quarto:status" content="draft">


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><div id="quarto-draft-alert" class="alert alert-warning"><i class="bi bi-pencil-square"></i>Draft</div>
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">digressions on (mouse) brain imaging</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/jasonlerch/blog"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Unbalanced Four Core Genotype Analyses</h1>
                  <div>
        <div class="description">
          You want to run a four core genotype analysis without having all four genotypes
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">R</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Jason Lerch and Kamila Szulc-Lerch </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 8, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<p>This came out of a discussion related to a rat model designed to explore the same question at the Four Core Genotype mouse model: what happens if you don’t have all four arms? Are you hopelessly confounded? The short answer is no under some assumptions. We’ll explore that with some simple simulations.</p>
<p>To start with we’ll create a function to simulate the four core genotype model and then drop one of the arms.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>generateFCGData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">XXM =</span> <span class="dv">0</span>,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">XXF =</span> <span class="dv">0</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">XYM =</span> <span class="dv">0</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">XYF =</span> <span class="dv">0</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">std =</span> <span class="dv">1</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ngroup=</span><span class="dv">10</span>) {</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XXM"</span>, <span class="at">gonad=</span><span class="st">"T"</span>, <span class="at">chromosomes=</span><span class="st">"XX"</span>, </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XXM, <span class="at">sd=</span>std)),</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XXF"</span>, <span class="at">gonad=</span><span class="st">"O"</span>, <span class="at">chromosomes=</span><span class="st">"XX"</span>, </span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XXF, <span class="at">sd=</span>std)),</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XYM"</span>, <span class="at">gonad=</span><span class="st">"T"</span>, <span class="at">chromosomes=</span><span class="st">"XY"</span>, </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XYM, <span class="at">sd=</span>std)),</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XYF"</span>, <span class="at">gonad=</span><span class="st">"O"</span>, <span class="at">chromosomes=</span><span class="st">"XY"</span>, </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XYF, <span class="at">sd=</span>std))</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s assume that having male gonads increases the output variable by 2 standard deviations. That would look as follows:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>ddd <span class="ot">&lt;-</span> <span class="fu">generateFCGData</span>(<span class="at">XYM=</span><span class="dv">2</span>, <span class="at">XXM=</span><span class="dv">2</span>, <span class="at">ngroup =</span> <span class="dv">500</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ddd) <span class="sc">+</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>volume, <span class="at">fill=</span>chromosomes) <span class="sc">+</span> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gonad, <span class="at">scales=</span><span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span> </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Let’s run the stats three ways, first separately fitting gonads and chromosomes and then using an additive model incorporating both chromosomes and gonads:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lfg <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7156 -0.6636 -0.0165  0.6493  4.7112 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.009019   0.031619   0.285    0.775    
gonadT      2.005040   0.044716  44.839   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9999 on 1998 degrees of freedom
Multiple R-squared:  0.5016,    Adjusted R-squared:  0.5013 
F-statistic:  2011 on 1 and 1998 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lfc <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> chromosomes, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ chromosomes, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1828 -1.0775 -0.0058  1.0610  3.7782 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   1.009075   0.044786  22.531   &lt;2e-16 ***
chromosomesXY 0.004928   0.063337   0.078    0.938    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.416 on 1998 degrees of freedom
Multiple R-squared:  3.03e-06,  Adjusted R-squared:  -0.0004975 
F-statistic: 0.006054 on 1 and 1998 DF,  p-value: 0.938</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lfgc <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad + chromosomes, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7131 -0.6636 -0.0182  0.6503  4.7088 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   0.006555   0.038735   0.169    0.866    
gonadT        2.005040   0.044727  44.828   &lt;2e-16 ***
chromosomesXY 0.004928   0.044727   0.110    0.912    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1 on 1997 degrees of freedom
Multiple R-squared:  0.5016,    Adjusted R-squared:  0.5011 
F-statistic:  1005 on 2 and 1997 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<p>The output here is correct - no matter how analyzed, the statistics tell us that having testes increases the output by 2 (everything is in units of standard deviation here), and there is no effect of chromosomes.</p>
<p>Let’s now drop one of the arms, and assume that rather than having all four groups we do not have the XY female group. The exact same data would look like this:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ddd <span class="ot">&lt;-</span> ddd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">!=</span><span class="st">"XYF"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ddd) <span class="sc">+</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>volume, <span class="at">fill=</span>chromosomes) <span class="sc">+</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gonad, <span class="at">scales=</span><span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span> </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>And we run the same statistics:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lrg <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7156 -0.6517 -0.0149  0.6648  3.1468 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.00683    0.04411  -0.155    0.877    
gonadT       2.02089    0.05403  37.405   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9864 on 1498 degrees of freedom
Multiple R-squared:  0.4829,    Adjusted R-squared:  0.4826 
F-statistic:  1399 on 1 and 1498 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lrc <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> chromosomes, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ chromosomes, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7430 -0.9230 -0.0218  0.9341  3.6675 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    1.00908    0.04077   24.75   &lt;2e-16 ***
chromosomesXY  0.99406    0.07061   14.08   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.289 on 1498 degrees of freedom
Multiple R-squared:  0.1169,    Adjusted R-squared:  0.1163 
F-statistic: 198.2 on 1 and 1498 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( ( lrgc <span class="ot">&lt;-</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, ddd)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad + chromosomes, data = ddd)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7265 -0.6531 -0.0141  0.6660  3.1468 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   -0.00683    0.04413  -0.155    0.877    
gonadT         2.03181    0.06240  32.559   &lt;2e-16 ***
chromosomesXY -0.02184    0.06240  -0.350    0.726    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9867 on 1497 degrees of freedom
Multiple R-squared:  0.483, Adjusted R-squared:  0.4823 
F-statistic: 699.2 on 2 and 1497 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<p>Losing one arm of the study means that the design is now unbalanced. Looking at the three different stats models shows that modelling only gonads gives the correct answer of a 2 SD increase, but modelling only chromosomes gives an incorrect answer of an increase of 1SD when having XY chromosomes. What saves us is covarying - modelling gonads + chromosomes again returns the correct answer, since the model simultaneously estimates the effects of gonads and chromosomes while controlling for the other.</p>
<p>Let’s show that graphically:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>allStats <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lfg, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"FCG: G"</span>),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lfc, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"FCG: C"</span>),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lfgc, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"FCG: G + C"</span>),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lrg, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"uFCG: G"</span>),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lrc, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"uFCG: C"</span>),</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tidy</span>(lrgc, <span class="at">conf.int =</span> T) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"uFCG: G + C"</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">2</span>, <span class="st">"Set2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in RColorBrewer::brewer.pal(2, "Set2"): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>allStats <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x=</span>term, <span class="at">y=</span>estimate, <span class="at">colour=</span>term) <span class="sc">+</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>conf.low, <span class="at">ymax=</span>conf.high)) <span class="sc">+</span> </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>model, <span class="at">scale=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cols[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) <span class="sc">+</span> </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">2</span>, <span class="at">color=</span>cols[<span class="dv">2</span>]) <span class="sc">+</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color=</span>cols[<span class="dv">1</span>]) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Model effects"</span>, <span class="at">caption=</span><span class="st">"FCG = balanced four core genotypes; uFCG = unbalanced four core genotypes (no XYF); C = chromosomes; G = gonads"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now, can’t we just do a simple t-test comparing two groups with the desired contrasts which avoids the issue of more complex modelling? Yes, that works:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> group, <span class="at">data =</span> ddd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"XXF"</span>, <span class="st">"XXM"</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ group, data = ddd %&gt;% filter(group %in% 
    c("XXF", "XXM")))

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7265 -0.6673 -0.0127  0.6692  3.1468 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.00683    0.04414  -0.155    0.877    
groupXXM     2.03181    0.06242  32.551   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9869 on 998 degrees of freedom
Multiple R-squared:  0.515, Adjusted R-squared:  0.5145 
F-statistic:  1060 on 1 and 998 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<p>But do we pay a price in statistical power? Let’s compare the simple two group comparison to the linear model with all three groups (in the unbalanced FCG case) estimating both a gonad and a chromosome effect.</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>simNtimes <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">nsims=</span><span class="dv">100</span>, <span class="at">effectSize=</span><span class="dv">2</span>, <span class="at">ngroup=</span><span class="dv">20</span>) {</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  bFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGData</span>(<span class="at">XYM=</span>effectSize, </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">XXM=</span>effectSize, </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">ngroup=</span>ngroup))</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  uFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(bFCG, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group<span class="sc">!=</span><span class="st">"XYF"</span>))</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  tgroup <span class="ot">&lt;-</span> <span class="fu">map</span>(bFCG, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"XXF"</span>, <span class="st">"XXM"</span>)))</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  lbFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(bFCG, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  luFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(uFCG, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  ltgroup <span class="ot">&lt;-</span> <span class="fu">map</span>(tgroup, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> group, <span class="at">data=</span>.x))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  tlbFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(lbFCG, tidy)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  tluFCG <span class="ot">&lt;-</span> <span class="fu">map</span>(luFCG, tidy)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  tltgroup <span class="ot">&lt;-</span> <span class="fu">map</span>(ltgroup, tidy)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">bFCG =</span> <span class="fu">map_dbl</span>(tlbFCG, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"gonadT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">uFCG =</span> <span class="fu">map_dbl</span>(tluFCG, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"gonadT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">twogroup =</span> <span class="fu">map_dbl</span>(tltgroup, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"groupXXM"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value))</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>allSims <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="at">by=</span><span class="fl">0.1</span>), <span class="sc">~</span> <span class="fu">simNtimes</span>(<span class="at">nsims=</span><span class="dv">200</span>, <span class="at">effectSize=</span>.x, <span class="at">ngroup =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">d=</span>.x))</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>powerDF <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(allSims, <span class="sc">~</span> .x) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(d) <span class="sc">%&gt;%</span> <span class="fu">summarise_at</span>(<span class="fu">vars</span>(bFCG<span class="sc">:</span>twogroup), <span class="cf">function</span>(x) <span class="fu">mean</span>(x<span class="sc">&lt;</span><span class="fl">0.05</span>))</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>powerDF <span class="sc">%&gt;%</span> </span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(bFCG<span class="sc">:</span>twogroup, <span class="at">values_to =</span> <span class="st">"power"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">x=</span>d, <span class="at">y=</span>power, <span class="at">colour=</span>name) <span class="sc">+</span> </span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"gam"</span>, <span class="at">formula=</span>y<span class="sc">~</span><span class="fu">s</span>(x, <span class="at">k=</span><span class="dv">8</span>), <span class="at">se=</span>F) <span class="sc">+</span> </span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="st">"Model"</span>, <span class="at">palette =</span> <span class="st">"Dark2"</span>, </span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>                     <span class="at">label=</span><span class="fu">c</span>(<span class="st">"balanced FCG"</span>, </span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"unbalanced FCG"</span>, </span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"two group comparison"</span>)) <span class="sc">+</span> </span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Simulated effect size"</span>) <span class="sc">+</span> </span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Power of gonad effect"</span>, <span class="at">caption =</span><span class="st">"Assuming n=10 per group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Not really - this shows the power we lose when moving from a balanced FCG dataset to the unbalanced one missing one group. But the power is the same between the two ways of modelling the unbalanced dataset, indicating that in the case of the unbalanced design (i.e.&nbsp;only having three rather than all four groups of the FCG model) the gonad and chromosome terms of the statistical model essentially reduce down to a two-group comparison.</p>
<p>Let’s expand this to the FCG-like rat model (male arm only for now). There are four genotypes:</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Genotype</th>
<th>Gonads</th>
<th>Sex Chromosomes</th>
<th>Sry</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>XX</td>
<td>Ovaries</td>
<td>XX</td>
<td>None</td>
</tr>
<tr class="even">
<td>XX-Sry</td>
<td>Testes</td>
<td>XX</td>
<td>Transgene</td>
</tr>
<tr class="odd">
<td>XY</td>
<td>Testes</td>
<td>XY</td>
<td>WT</td>
</tr>
<tr class="even">
<td>XY-Sry</td>
<td>Testes</td>
<td>XY</td>
<td>WT + Transgene</td>
</tr>
</tbody>
</table>
<p>Once again we need a function to simulate from this dataset</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>generateFCGRatData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">XX =</span> <span class="dv">0</span>,</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">XXSry =</span> <span class="dv">0</span>,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">XY =</span> <span class="dv">0</span>,</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">XYSry =</span> <span class="dv">0</span>,</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">std =</span> <span class="dv">1</span>,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ngroup=</span><span class="dv">10</span>) {</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XX"</span>, <span class="at">gonad=</span><span class="st">"O"</span>, <span class="at">chromosomes=</span><span class="st">"XX"</span>, <span class="at">sry=</span><span class="dv">0</span>,</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XX, <span class="at">sd=</span>std)),</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XXSry"</span>, <span class="at">gonad=</span><span class="st">"T"</span>, <span class="at">chromosomes=</span><span class="st">"XX"</span>, <span class="at">sry=</span><span class="dv">1</span>,</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XXSry, <span class="at">sd=</span>std)),</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XY"</span>, <span class="at">gonad=</span><span class="st">"T"</span>, <span class="at">chromosomes=</span><span class="st">"XY"</span>, <span class="at">sry=</span><span class="dv">1</span>,</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XY, <span class="at">sd=</span>std)),</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group=</span><span class="st">"XYSry"</span>, <span class="at">gonad=</span><span class="st">"T"</span>, <span class="at">chromosomes=</span><span class="st">"XY"</span>, <span class="at">sry=</span><span class="dv">2</span>,</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">volume =</span> <span class="fu">rnorm</span>(ngroup, XYSry, <span class="at">sd=</span>std))</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s see what a few different scenarios look like:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>gonads <span class="ot">&lt;-</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>, <span class="at">XXSry=</span><span class="dv">2</span>, <span class="at">XY=</span><span class="dv">2</span>, <span class="at">XYSry=</span><span class="dv">2</span>, <span class="at">ngroup =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">contrast =</span> <span class="st">"Gonad"</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>chrom <span class="ot">&lt;-</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>, <span class="at">XXSry=</span><span class="dv">0</span>, <span class="at">XY=</span><span class="dv">2</span>, <span class="at">XYSry=</span><span class="dv">2</span>, <span class="at">ngroup =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">contrast =</span> <span class="st">"Chromosomes"</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>sry <span class="ot">&lt;-</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>, <span class="at">XXSry=</span><span class="dv">2</span>, <span class="at">XY=</span><span class="dv">2</span>, <span class="at">XYSry=</span><span class="dv">4</span>, <span class="at">ngroup =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">contrast =</span> <span class="st">"Sry"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(gonads, chrom, sry) <span class="sc">%&gt;%</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>volume, <span class="at">fill=</span>chromosomes) <span class="sc">+</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gonad, <span class="at">scales=</span><span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span> </span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(contrast <span class="sc">~</span> .) <span class="sc">+</span> </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Can we appropriately detect the right effects in those models, focusing first on analyzing gonads and chromosomes?</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>gonads))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad + chromosomes, data = gonads)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7327 -0.7017  0.0092  0.6782  3.4847 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   -0.01110    0.04458  -0.249    0.803    
gonadT         1.99798    0.06305  31.687   &lt;2e-16 ***
chromosomesXY  0.03613    0.05461   0.662    0.508    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.997 on 1997 degrees of freedom
Multiple R-squared:  0.4359,    Adjusted R-squared:  0.4353 
F-statistic: 771.6 on 2 and 1997 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>chrom))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ gonad + chromosomes, data = chrom)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.2552 -0.6764  0.0075  0.6762  3.9444 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   -0.0002106  0.0452218  -0.005    0.996    
gonadT         0.0081282  0.0639532   0.127    0.899    
chromosomesXY  2.0090115  0.0553851  36.273   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.011 on 1997 degrees of freedom
Multiple R-squared:  0.4981,    Adjusted R-squared:  0.4976 
F-statistic: 990.8 on 2 and 1997 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<p>Yes. What about treating sry as an ordered factor (i.e.&nbsp;assuming that a dose of 2 is more than a dose of 1, but not assuming that it is purely linear).</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>gonads <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sry=</span><span class="fu">as.ordered</span>(sry))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ sry + chromosomes, data = gonads %&gt;% mutate(sry = as.ordered(sry)))

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7327 -0.6991  0.0081  0.6783  3.4847 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    1.31744    0.03933  33.497   &lt;2e-16 ***
sry.L          1.40547    0.06307  22.285   &lt;2e-16 ***
sry.Q         -0.81990    0.03641 -22.517   &lt;2e-16 ***
chromosomesXY  0.04130    0.06307   0.655    0.513    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9972 on 1996 degrees of freedom
Multiple R-squared:  0.4359,    Adjusted R-squared:  0.4351 
F-statistic: 514.1 on 3 and 1996 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>chrom <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sry=</span><span class="fu">as.ordered</span>(sry))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ sry + chromosomes, data = chrom %&gt;% mutate(sry = as.ordered(sry)))

Residuals:
    Min      1Q  Median      3Q     Max 
-3.2552 -0.6742  0.0073  0.6762  3.9444 

Coefficients:
               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   -0.004533   0.039890  -0.114    0.910    
sry.L         -0.014917   0.063966  -0.233    0.816    
sry.Q         -0.015249   0.036931  -0.413    0.680    
chromosomesXY  2.023624   0.063966  31.636   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.011 on 1996 degrees of freedom
Multiple R-squared:  0.4981,    Adjusted R-squared:  0.4974 
F-statistic: 660.4 on 3 and 1996 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>sry <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sry=</span><span class="fu">as.ordered</span>(sry))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = volume ~ sry + chromosomes, data = sry %&gt;% mutate(sry = as.ordered(sry)))

Residuals:
    Min      1Q  Median      3Q     Max 
-3.5094 -0.6478 -0.0062  0.6706  3.6643 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   1.994594   0.038950  51.209   &lt;2e-16 ***
sry.L         2.863991   0.062459  45.854   &lt;2e-16 ***
sry.Q         0.033932   0.036061   0.941    0.347    
chromosomesXY 0.003616   0.062459   0.058    0.954    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.9876 on 1996 degrees of freedom
Multiple R-squared:  0.6786,    Adjusted R-squared:  0.6781 
F-statistic:  1405 on 3 and 1996 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<p>Still works but is a bit more unstable. Some interpretation aids to these models - when treating sry as an ordered factor, there are two possible trends - a linear trends, indicating that with every increase in sry there is an increase in the outcome, and a quadratic trend, indicating that going from 1 to 2 is not the same as going from 0 to 1. So in our third model where we artificially simulated an increase with every dose of sry there is a linear increase (sry.L), but no quadratic increase (sry.Q is non-significant). But in the gonad model there is both a linear and a quadratic effect, since having 2 sry copies (WT + transgene) has the same effect as having one sry copy.</p>
<p>What about power? First just looking at the gonads + chromosomes model:</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>simRatNtimes <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">nsims=</span><span class="dv">100</span>, <span class="at">effectSize=</span><span class="dv">2</span>, <span class="at">ngroup=</span><span class="dv">20</span>) {</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  ratG <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XXSry=</span>effectSize,</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XY=</span>effectSize,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XYSry=</span>effectSize,</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">ngroup=</span>ngroup))</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  ratC <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>,</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XXSry=</span><span class="dv">0</span>,</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XY=</span>effectSize,</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XYSry=</span>effectSize,</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">ngroup=</span>ngroup))</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  lratG <span class="ot">&lt;-</span> <span class="fu">map</span>(ratG, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  lratC <span class="ot">&lt;-</span> <span class="fu">map</span>(ratC, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> gonad <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  tlratG <span class="ot">&lt;-</span> <span class="fu">map</span>(lratG, tidy)</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  tlratC <span class="ot">&lt;-</span> <span class="fu">map</span>(lratC, tidy)</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">gonads =</span> <span class="fu">map_dbl</span>(tlratG, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"gonadT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosomes =</span> <span class="fu">map_dbl</span>(tlratC, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"chromosomesXY"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value))</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>allRatSims <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="at">by=</span><span class="fl">0.1</span>), <span class="sc">~</span> <span class="fu">simRatNtimes</span>(<span class="at">nsims=</span><span class="dv">200</span>, <span class="at">effectSize=</span>.x, <span class="at">ngroup =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">d=</span>.x))</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>powerDF <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(allRatSims, <span class="sc">~</span> .x) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(d) <span class="sc">%&gt;%</span> <span class="fu">summarise_at</span>(<span class="fu">vars</span>(gonads<span class="sc">:</span>chromosomes), <span class="cf">function</span>(x) <span class="fu">mean</span>(x<span class="sc">&lt;</span><span class="fl">0.05</span>))</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>powerDF <span class="sc">%&gt;%</span> </span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(gonads<span class="sc">:</span>chromosomes, <span class="at">values_to =</span> <span class="st">"power"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">x=</span>d, <span class="at">y=</span>power, <span class="at">colour=</span>name) <span class="sc">+</span> </span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"gam"</span>, <span class="at">formula=</span>y<span class="sc">~</span><span class="fu">s</span>(x, <span class="at">k=</span><span class="dv">8</span>), <span class="at">se=</span>F) <span class="sc">+</span> </span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="st">"Model"</span>, <span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> </span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Simulated effect size"</span>) <span class="sc">+</span> </span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Power of FCG-like rat study"</span>, </span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle=</span><span class="st">"Modelling gonads + chromosomes"</span>,</span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span><span class="st">"Assuming n=10 per group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Given the unbalanced design for gonads there is some power differential, in that it is easier to detect a chromosome effect than a gonad effect.</p>
<p>And if we model sry as an ordered factor?</p>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>simRatNtimesOF <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">nsims=</span><span class="dv">100</span>, <span class="at">effectSize=</span><span class="dv">2</span>, <span class="at">ngroup=</span><span class="dv">20</span>) {</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  ratG <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XXSry=</span>effectSize,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XY=</span>effectSize,</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XYSry=</span>effectSize,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">ngroup=</span>ngroup) <span class="sc">%&gt;%</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sry =</span> <span class="fu">as.ordered</span>(sry))) </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  ratC <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>,</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XXSry=</span><span class="dv">0</span>,</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XY=</span>effectSize,</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XYSry=</span>effectSize,</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">ngroup=</span>ngroup) <span class="sc">%&gt;%</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sry =</span> <span class="fu">as.ordered</span>(sry))) </span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  ratS <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>nsims, <span class="sc">~</span> <span class="fu">generateFCGRatData</span>(<span class="at">XX=</span><span class="dv">0</span>,</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XXSry=</span>effectSize,</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XY=</span>effectSize,</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">XYSry=</span>effectSize<span class="sc">*</span><span class="dv">2</span>,</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">ngroup=</span>ngroup) <span class="sc">%&gt;%</span></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sry =</span> <span class="fu">as.ordered</span>(sry))) </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>  lratG <span class="ot">&lt;-</span> <span class="fu">map</span>(ratG, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>  lratC <span class="ot">&lt;-</span> <span class="fu">map</span>(ratC, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  lratS <span class="ot">&lt;-</span> <span class="fu">map</span>(ratS, <span class="sc">~</span> <span class="fu">lm</span>(volume <span class="sc">~</span> sry <span class="sc">+</span> chromosomes, <span class="at">data=</span>.x))</span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a>  tlratG <span class="ot">&lt;-</span> <span class="fu">map</span>(lratG, tidy)</span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a>  tlratC <span class="ot">&lt;-</span> <span class="fu">map</span>(lratC, tidy)</span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>  tlratS <span class="ot">&lt;-</span> <span class="fu">map</span>(lratS, tidy)</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">gonads =</span> <span class="fu">map_dbl</span>(tlratG, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"sry.L"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosomes =</span> <span class="fu">map_dbl</span>(tlratC, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"chromosomesXY"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">sryL =</span> <span class="fu">map_dbl</span>(tlratS, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"sry.L"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value)),</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">sryQ =</span> <span class="fu">map_dbl</span>(tlratS, <span class="sc">~</span>.x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"sry.Q"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(p.value))</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a>allRatSims <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="at">by=</span><span class="fl">0.1</span>), <span class="sc">~</span> <span class="fu">simRatNtimesOF</span>(<span class="at">nsims=</span><span class="dv">200</span>, <span class="at">effectSize=</span>.x, <span class="at">ngroup =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">d=</span>.x))</span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a>powerDF <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(allRatSims, <span class="sc">~</span> .x) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(d) <span class="sc">%&gt;%</span> <span class="fu">summarise_at</span>(<span class="fu">vars</span>(gonads<span class="sc">:</span>sryQ), <span class="cf">function</span>(x) <span class="fu">mean</span>(x<span class="sc">&lt;</span><span class="fl">0.05</span>))</span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-41"><a href="#cb36-41" aria-hidden="true" tabindex="-1"></a>powerDF <span class="sc">%&gt;%</span> </span>
<span id="cb36-42"><a href="#cb36-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(gonads<span class="sc">:</span>sryQ, <span class="at">values_to =</span> <span class="st">"power"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-43"><a href="#cb36-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">x=</span>d, <span class="at">y=</span>power, <span class="at">colour=</span>name) <span class="sc">+</span> </span>
<span id="cb36-44"><a href="#cb36-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"gam"</span>, <span class="at">formula=</span>y<span class="sc">~</span><span class="fu">s</span>(x, <span class="at">k=</span><span class="dv">8</span>), <span class="at">se=</span>F) <span class="sc">+</span> </span>
<span id="cb36-45"><a href="#cb36-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="st">"Model"</span>, <span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> </span>
<span id="cb36-46"><a href="#cb36-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb36-47"><a href="#cb36-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Simulated effect size"</span>) <span class="sc">+</span> </span>
<span id="cb36-48"><a href="#cb36-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Power of FCG-like rat study"</span>, </span>
<span id="cb36-49"><a href="#cb36-49" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle=</span><span class="st">"Modelling sry + chromosomes"</span>,</span>
<span id="cb36-50"><a href="#cb36-50" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span><span class="st">"Assuming n=10 per group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Here the <em>gonads</em> model looks at the linear part of the sry term when there is a change in the presence of sry but no dosage effect. <em>sryL</em> looks at the linear term when there is a dosage effect, which is now the most powered term (since there are three possible steps), and <em>sryQ</em> looks at the quadratic effect when there is a simulated dosage effect which is rightly 0.</p>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/users\.ox\.ac\.uk\/~ndcn0890\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://utteranc.es/client.js" repo="jasonlerch/blogComments" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">

<div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>